### PR TITLE
Support custom URL for Substack pages

### PR DESCRIPTION
**Pull request by ArtemKolodko**

### CODE REVIEW

File: server/middlewares/substack.ts [hunk 1]
```diff
@@ -7,19 +7,23 @@ const client = buildClient()
 
 const substack = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
   const host = req.get('host')
+  let pageUrl = req.query.url as string || ''
```
Comment (polymorpher) [inline server/middlewares/substack.ts]: This was initially proposed, but changed later, because anyone would be able to use the server as a proxy and spam

**Comment by polymorpher** [comment]
Add an environment variable (UNRESTRICTED_PROXY) to active this only when it is present?

File: server/middlewares/substack.ts (hunk 1 above)
Comment (ArtemKolodko) [reply server/middlewares/substack.ts]: I think simple rate limiter on application side might be enough at first

Comment (ArtemKolodko) [reply server/middlewares/substack.ts]: Checked the code, you already added it

**Comment by ArtemKolodko** [comment]
Added UNRESTRICTED_PROXY env (1, true)

**Review by polymorpher — APPROVED** [review]

**Pull request merged**